### DISCUSSION TITLE
v5 performance

### DISCUSSION THREAD

**Discussion by jyuloeng** [TanStack/query#6068] [General]
Does v5 version improve performance compared with v4? How much is it?

What is the purpose of the upgrade of v5 version?

**Comment by frederikhors** [comment]
Yes. Please. This is an answer we need to convince the team to upgrade. And also out of curiosity.

Are there metrics and ways to measure these metrics?

**Reply by jyuloeng** [reply]
ummm... I just think some performance improvements in concurrency or resource consumption do not seem to be explained?

I can't find any reason to upgrade to v5 in Migrating to TanStack Query v5.

Please tell me how upgrading to v5 can improve my use. Thanks:D

**Comment by DamianOsipiuk** [comment]
Not sure why people expect `major` version in the  `library` to be something special. 
It's not a product, it's utility for developers.

New `major` version is:
- a way to fix some mistakes made during development that result in breaking changes
- remove deprecated functionality
- introduce new features that are incopatible with previous implementation, or change old behavior

It's totally up to you to make a decision if you want to update to latest version. Not always you will get an incentive to do so.
Most wanted stuff and performance improvements can be done in `minor` versions, we don't have to wait for a `major` one.

If you try to update in `beta` or `rc` phase you are helping us catch potential bugs and fix them before final release goes out.
Also you can give us feedback about the changes we made that might not work as expected. We would still have time to rethink them or even revert, which would be impossible after release goes out.
It's a way to give back to the community, who works together to bring you this library for free.

If you decide not to upgrade - it's totally fine.
Just keep in mind that old versions would receive only important bugfixes.
So eventually you would want to reconsider as you will fall behind and upgrade might be more tedious later on. Especially if you would be few major versions behind.

**Reply by frederikhors** [reply]
Thank you. But I only meant to say that perhaps having more specific information (metrics) on the performance differences (v4 - v5) would be useful to push those who decide in the team to update FIRST: everything you wrote is obvious; What's less obvious is the short upgrade time on teams as big and slow as turtles.

**Reply by DamianOsipiuk** [reply]
It would argue that overhead of `@tanstack/query` is quite marginal. So any benchmark would have to be very unnatural.
What we could emphasis more though is `bundle size` reduction, but people using `bundleophobia` would not notice that, as 5.0 introduced additional hooks.

**Reply by jyuloeng** [reply]
@DamianOsipiuk  Thanks for your answer. I'm also a fan of `@tanstack/query`, and my team uses a lot of `useQuery` in our project, but I find a lot of break-change from 《Migrating to TanStack Query v5》, these changes may pose a risk to our production environment, so I also hope more specific information (metrics) on the performance differences (v4 - v5) would be useful to push those who decide in the team to update. 

Thanks again~

**Comment by TkDodo** [comment]
> How much is it?

apparently, it's "so much faster"

https://x.com/okandass/status/1707649874924114080
